src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -1578,6 +1601,24 @@ private boolean isUserValid(final RegisteredUser userToValidate) {
         return isValid;
     }
 
+    /**
+     * This function checks that the name provided is valid.
+     *
+     * @param name
+     *            - the name to validate.
+     * @return true if the name is valid, false otherwise.
+     */
+    public boolean isUserNameValid(final String name){
+        boolean isValid = true;
+
+        Pattern illegalCharacters = Pattern.compile(USER_NAME_ILLEGAL_CHARS_REGEX);
+        if (name.length() > USER_NAME_MAX_LENGTH || illegalCharacters.matcher(name).find()) {
+            isValid = false;
```

Comment:
It might be a little nicer to have this as `return false;` and line 1619 as `return true;` removing the need for line 1612 - which matches most of the boolean returns we have, though the method above also doesn't follow this style.